mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -566,8 +582,8 @@ class Word2VecModel private[spark] (
     wordList.zip(cosVec)
       .toSeq
       .sortBy(-_._2)
-      .take(num + 1)
-      .tail
+      .filter(tup => wordOpt.map(w => !w.equals(tup._1)).getOrElse(true) && tup._2 != 1.0d)
```

Comment:
@hhbyyh So [one of the python doctests](https://github.com/apache/spark/blob/branch-2.0/python/pyspark/mllib/feature.py#L575) depends on a word not being a synonym of its vector representation.  I think since this is the documented behavior now, that's the direction the fix should go as well, but I'll use `Vector.equals` instead of checking similarity in any case.